src/execution_plan/ops/op_expand_into.c

```diff
@@ -6,68 +6,270 @@
+ * removed filter matrix from original expression
+ * clears filter matrix. */
+static void _traverse(OpExpandInto *op) {
+    // Append filter matrix to algebraic expression, as the right most operand.
+    AlgebraicExpression_AppendTerm(op->ae, op->F, false, false);
+
+    // Evaluate expression.
+    AlgebraicExpression_Execute(op->ae, op->M);
+    
+    // Remove operand.
+    AlgebraicExpression_RemoveTerm(op->ae, op->ae->operand_count-1, NULL);
+
+    // Clear filter matrix.
+    GrB_Matrix_clear(op->F);
+}
+
+OpBase* NewExpandIntoOp(AlgebraicExpression *ae, AST *ast) {
+    OpExpandInto *expandInto = calloc(1, sizeof(OpExpandInto));
+    GraphContext *gc = GraphContext_GetFromTLS();
+    expandInto->ae = ae;
+    expandInto->F = NULL;
+    expandInto->r = NULL;
+    expandInto->ast = ast;
+    expandInto->edges = NULL;
+    expandInto->graph = gc->g;
+    expandInto->recordsLen = 0;
+    expandInto->transposed_edge = false;
+    expandInto->edgeRelationTypes = NULL;
+    expandInto->recordsCap = _determinRecordCap(ast);
+    expandInto->srcNodeRecIdx = AST_GetAliasID(ast, ae->src_node->alias);
+    expandInto->destNodeRecIdx = AST_GetAliasID(ast, ae->dest_node->alias);
+    expandInto->records = rm_calloc(expandInto->recordsCap, sizeof(Record));
 
     // Set our Op operations
     OpBase_Init(&expandInto->op);
     expandInto->op.name = "Expand Into";
     expandInto->op.type = OPType_EXPAND_INTO;
-    expandInto->op.consume = OpExpandIntoConsume;
-    expandInto->op.reset = OpExpandIntoReset;
-    expandInto->op.free = OpExpandIntoFree;
-    expandInto->op.modifies = NewVector(char*, 0);
+    expandInto->op.init = ExpandIntoInit;
+    expandInto->op.free = ExpandIntoFree;
+    expandInto->op.reset = ExpandIntoReset;
+    expandInto->op.consume = ExpandIntoConsume;
+    expandInto->op.toString = ExpandIntoToString;
+    expandInto->op.modifies = NULL;
+
+    if(ae->edge) {
+        char *modified = NULL;
+        _setupTraversedRelations(expandInto, gc);
+        modified = expandInto->ae->edge->alias;
+        expandInto->op.modifies = NewVector(char*, 0);
+        Vector_Push(expandInto->op.modifies, modified);
+        expandInto->edges = array_new(Edge, 32);
+        expandInto->edgeRecIdx = AST_GetAliasID(ast, ae->edge->alias);
+    }
+
     return (OpBase*)expandInto;
 }
 
-Record OpExpandIntoConsume(OpBase *opBase) {
+OpResult ExpandIntoInit(OpBase *opBase) {
+    OpExpandInto *op = (OpExpandInto*)opBase;
+
+    size_t required_dim = Graph_RequiredMatrixDim(op->graph);
+    GrB_Matrix_new(&op->M, GrB_BOOL, required_dim, op->recordsCap);
+    GrB_Matrix_new(&op->F, GrB_BOOL, required_dim, op->recordsCap);
+
+    size_t op_idx = op->ae->operand_count - 1;
+    AlgebraicExpression *exp = op->ae;
+    // If the input is set to be transposed on the first expression evaluation,
+    // the source and destination nodes will be swapped in the record.
+    op->transposed_edge = exp->edge && exp->operands[op_idx].transpose;
```

Comment:
`op->transposed_edge` is unused. Is this op responsible for src/dest swapping?

Reply:
👍